### PR TITLE
boards: nrf9160dk: Change default revision to 0.14.0

### PR DESCRIPTION
**Pull request by gregersrygg**
Change the default revision of the nRF9160 DK to 0.14.0 to make it easier to use external flash. The v0.14.0 revision is almost 3 years old.

It will still be possible to build without external flash support for older boards when specifying a revision older than v0.14.0:

```bash
west build -b nrf9160dk_nrf9160_ns@0.7.0
```

### CODE REVIEW

**Comment by anangl** [comment]
> Change the default revision of the nRF9160 DK to 0.14.0 to make it easier to use external flash. The v0.14.0 revision is almost 3 years old.

I think that sounds reasonable. But first we need to apply corrections from #58136 so that CI for this PR has a chance to become green.

**Comment by anangl** [comment]
@gregersrygg Could you also add a related release note?

File: doc/releases/release-notes-3.4.rst [hunk 1]
```diff
@@ -305,6 +305,13 @@ Boards & SoC Support
 
 * Made these changes for ARM boards:
 
+  * Changed the default board revision to 0.14.0 for the Nordic boards
+    ``nrf9160dk_nrf9160`` and ``nrf9160dk_nrf52840``. To build for an
+    older revision of the nrf9160dk without external flash, specify an
+    older board revision when building.
+
+  * Enable external_flash_pins_routing switch in ``nrf9160dk_nrf52840`` by default.
```
Comment (anangl) [inline doc/releases/release-notes-3.4.rst:313]:
```suggestion
  * Enabled external_flash_pins_routing switch in ``nrf9160dk_nrf52840`` by default.
```

File: doc/releases/release-notes-3.4.rst [hunk 2]
```diff
@@ -305,6 +305,13 @@ Boards & SoC Support
 
 * Made these changes for ARM boards:
 
+  * Changed the default board revision to 0.14.0 for the Nordic boards
+    ``nrf9160dk_nrf9160`` and ``nrf9160dk_nrf52840``. To build for an
+    older revision of the nrf9160dk without external flash, specify an
+    older board revision when building.
```
Comment (anangl) [inline doc/releases/release-notes-3.4.rst:311]:
```suggestion
    older revision of the nRF9160 DK without external flash, specify that
    older board revision when building.
```

**Review by anangl — DISMISSED** [review]

**Review by anangl — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Pull request merged**